actors/builtin/storage_miner/storage_miner_actor_state.go

```diff
@@ -14,9 +15,12 @@ import (
 // Balance of a StorageMinerActor should equal exactly the sum of PreCommit deposits
 // that are not yet returned or burned.
 type StorageMinerActorState struct {
+	PreCommittedSectors PreCommittedSectorsAMT 
 	Sectors    SectorsAMT
+	FaultSet SectorNumberSetHAMT // TODO bitfield
+	ProvingSet cid.Cid
```

Comment:
Please document what it's a cid of, e.g. `HAMT[sectorNumber]sectorInfo`. Same for the HAMTs and AMTs above.

Until implementing the real structure, it's probably more useful to use a map or slice so that the desired programming semantics can be expressed in code.

Reply:
documented in comments, AMTs above have been defined in types.